state/protocol/inmem/dkg.go

```diff
@@ -14,7 +15,15 @@ type DKG flow.EpochCommit
 
 var _ protocol.DKG = (*DKG)(nil)
 
-func NewDKG(commit *flow.EpochCommit) *DKG { return (*DKG)(commit) }
+// NewDKG creates a new DKG instance from the given setup and commit events.
+// TODO(EFM, #6794): Remove this once we complete the network upgrade, we should remove v0 model.
```

Comment:
when I read this the first time, I thought you meant removing the _entrie_ constructor. Though, wouldn't we just remove the `commit.DKGIndexMap == nil` branch?

```suggestion
// TODO(EFM, #6794): Remove branch `commit.DKGIndexMap == nil` once we complete the network upgrade.
```